scripts/gendocs/generate

```diff
@@ -142,21 +144,22 @@ $output
 EOF
 }
 
-genclireference "docs" "cmd/acmesolver" "acmesolver"
-genclireference "docs" "cmd/cainjector" "cainjector"
-genclireference "docs" "cmd/ctl" "cmctl"
-genclireference "docs" "cmd/controller" "controller"
-genclireference "docs" "cmd/webhook" "webhook"
-
 # The branches named here exist in the `jetstack/cert-manager` repo.
 
-genversion "release-1.7" "next-docs"
+genversionwithcli "release-1.7" "next-docs"
 
 # In cert-manager 1.6 cert-manager.io and acme.cert-manager.io alpha and beta
 # API versions are no longer served, but the apis are still in the codebase. We
 # don't want to show the docs for those API versions so this is a workaround.
 # This will only be necessary for the release-1.6 branch.
 checkout "release-1.6"
```

Comment:
This `checkout` and the comment above relate to the `rm` and `gendocs` command below,
so I think it the `genclireference` commands should be moved after those.

Reply:
Good point, I have moved the comment.

`genclireference` must happen before we delete the legacy API versions because the code does not compile without them. I added a comment on that as well